tests/unit/test_testing/test_request_factory.py

```diff
@@ -81,6 +76,20 @@ async def test_request_factory_create_with_data(data_cls: DataContainerType) ->
     assert json.loads(body) == person_data
 
 
+async def test_request_factory_create_with_data_with_custom_encoder(mocker: MockerFixture) -> None:
```

Comment:
Instead of testing with different modelling libraries (for which the serialisation functionality is already tested elsewhere), what we actually want to test here is that the `RequestFactory` picks up the correct `type_encoders`.